keylime/da/record.py

```diff
@@ -0,0 +1,375 @@
+        self.tmp_d_cl = True
+        self.key_tls_priv = None
+        self.key_tls_pub = key_tls_pub
+        self.priv_key = None
+
+    def record_create(self, agent_data, attestation_data, ima_policy_data, service="auto", signed_attributes="auto"):
+        _ = (agent_data, attestation_data, ima_policy_data, service, signed_attributes)
+
+    def agent_list_retrieval(self, agent_list, record_prefix="auto", service="auto"):
+        _ = (agent_list, record_prefix, service)
+
+    def bulk_record_retrieval(self, record_identifier, record_list):
+        _ = (record_identifier, record_list)
+
+    def build_key_list(self, agent_identifier, aik_list, service="auto"):
+        _ = (agent_identifier, aik_list, service)
+
+    def record_read(self, attestation_record_list, agent_identifier, service="auto"):
+        _ = (attestation_record_list, agent_identifier, service)
+
+    def _get_record_type(self, service):
+        if service != "auto":
+            self.svc = service
+
+        if self.svc == "registrar":
+            return "registration"
+
+        return "attestation"
+
+    def _get_certs_path(self, override_service=None):
+
+        if override_service:
+            self.svc = override_service
+
+        if not self.key_tls_pub:
+            (_, self.key_tls_priv, _, _), _ = web_util.get_tls_options(self.svc, logger=logger)
+
+            self.key_tls_pub = self.key_tls_priv.replace("-private", "-public")
+
+            if self.tl_url:
+                with open(self.key_tls_priv, "rb") as fp:
+                    self.priv_key = crypto.rsa_import_privkey(fp.read())
+
+    def _record_encode(self, record_object):
+        """Encodes record"""
+
+        _encoded_record_object = record_object
+
+        if self.rcd_fmt == "pickle":
+            _encoded_record_object = pickle.dumps(record_object)
+
+        if self.rcd_fmt == "json":
+            _encoded_record_object = json.dumps(record_object, indent=4)
+            if self.rcd_enc == "base64":
+                _encoded_record_object = str.encode(_encoded_record_object)
+
+        if self.rcd_enc == "base64":
+            _encoded_record_object = base64.b64encode(_encoded_record_object)
+
+        return _encoded_record_object
+
+    def _record_decode(self, record_object):
+        """Decodes record"""
+
+        _decoded_record_object = record_object
+
+        if self.rcd_enc == "base64":
+            _decoded_record_object = base64.b64decode(_decoded_record_object)
+
+        if self.rcd_fmt == "pickle":
+            _decoded_record_object = pickle.loads(_decoded_record_object)
+
+        if self.rcd_fmt == "json":
+            if self.rcd_enc == "base64":
+                _decoded_record_object = _decoded_record_object.decode("utf-8")
+            _decoded_record_object = json.loads(_decoded_record_object)
+
+        return _decoded_record_object
+
+    def _record_sanitize(self, record_object):
```

Comment:
this is changing the parameter, should be cloned and returned the sanitiezed one

Reply:
This still needs to be adressed.